openedx_learning/apps/authoring/components/api.py

```diff
@@ -363,3 +387,171 @@ def create_component_version_content(
         learner_downloadable=learner_downloadable,
     )
     return cvrc
+
+
+class AssetError(StrEnum):
+    """Error codes related to fetching ComponentVersion assets."""
+    ASSET_PATH_NOT_FOUND_FOR_COMPONENT_VERSION = auto()
+    ASSET_NOT_LEARNER_DOWNLOADABLE = auto()
+    ASSET_HAS_NO_DOWNLOAD_FILE = auto()
+
+
+def _get_component_version_info_headers(component_version: ComponentVersion) -> dict[str, str]:
+    """
+    These are the headers we can derive based on a valid ComponentVersion.
+
+    These headers are intended to ease development and debugging, by showing
+    where this static asset is coming from. These headers will work even if
+    the asset path does not exist for this particular ComponentVersion.
+    """
+    component = component_version.component
+    learning_package = component.learning_package
+    return {
+        # Component
+        "X-Learning-Core-Component-Key": component.publishable_entity.key,
+        "X-Learning-Core-Component-Uuid": component.uuid,
+        # Component Version
+        "X-Learning-Core-Component-Version-Uuid": component_version.uuid,
+        "X-Learning-Core-Component-Version-Num": component_version.version_num,
+        # Learning Package
+        "X-Learning-Core-Learning-Package-Key": learning_package.key,
+        "X-Learning-Core-Learning-Package-Uuid": learning_package.uuid,
+    }
+
+
+def get_redirect_response_for_component_asset(
+    component_version_uuid: UUID,
+    asset_path: Path,
+    public: bool = False,
+    learner_downloadable_only: bool = True,
+) -> HttpResponse:
+    """
+    ``HttpResponse`` for a reverse-proxy to serve a ``ComponentVersion`` asset.
+
+    :param component_version_uuid: ``UUID`` of the ``ComponentVersion`` that the
+        asset is part of.
+
+    :param asset_path: Path to the asset being requested.
+
+    :param public: Is this asset going to be made available without auth checks?
+        If ``True``, this will return an ``HttpResponse`` that can be cached in
+        a CDN and shared across many clients.
+
+    :param learner_downloadable_only: Only return assets that are meant to be
+        downloadable by Learners, i.e. in the LMS experience. If this is
+        ``True``, then requests for assets that are not meant for student
+        download will return a ``404`` error response.
+
+    **Response Codes**
+
+    If the asset exists for this ``ComponentVersion``, this function will return
+    an ``HttpResponse`` with a status code of ``200``.
+
+    If the specified asset does not exist for this ``ComponentVersion``, or if
+    the ``ComponentVersion`` itself does not exist, the response code will be
+    ``404``.
+
+    *This function does not do auth checking of any sort*–it will never return
+    a ``401`` or ``403`` response code. That is by design. Figuring out who is
+    making the request and whether they have permission to do so is the
+    responsiblity of whatever is calling this function.
```

Comment:
That said, the `learner_downloadable_only` does seem like a type of auth check to me.

Reply:
Yeah, I feel a bit weird about that flag in general, but it's mostly there to try to keep people from shooting themselves in the foot when they're adding random things like LaTeX files and such–and to avoid issues we've had in the past around accidental downloads we've had in the past (e.g. `python-lib.zip`). So I thought it'd be better if the entire endpoint would make those assets unfindable/servable, and not just leave it up to the authz code.

Reply:
I think it's a solid design choice. I don't feel strongly, but maybe just reword the comment to be clear about it:

```
    Other than checking the coarse-grained ``learner_downloadable_only`` flag,
    *this function does not do auth checking*–it will never return a ``401`` or ``403``
    response code. That is by design. Figuring out who is making the request and
    whether they have permission to do so is the responsiblity of whatever is calling
    this function.
```